### PR TITLE
feat(wallet): Built out NFT Details Page UI

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Built out the NFT Details Page UI
(note: We do not yet have an API to retrieve NFT Metadata, the purpose of this PR is to get the UI ready for when an API is ready. Until then the NFT view is disabled.)

1) Built out the `nft-details` component
2) Refactored the `Portfolio` view to cut down on clutter and confusion.
- Portfolio's `accounts-and-transactions-list` is now it's own component.
- Portfolio's `token-lists` is now it's own component.

Resolves <https://github.com/brave/brave-browser/issues/20319>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

Demo:

https://user-images.githubusercontent.com/40611140/149214050-4365e9a0-d934-4d96-ae4a-ae59ee246da8.mov

### CODE REVIEW

**Review by onyb — COMMENTED** [review]
Since NFTs are non-fungible, I believe we'll always have only one account to display here. Just a thought that we can simplify the UX here.

Screenshot 2022-01-13 at 5 49 47 PM

**Comment by Douglashdaniel** [comment]
> Member
@onyb Updated this, also added a check to make sure we only show the `owner/account` with a balance!

Screen Shot 2022-01-13 at 9 13 44 AM

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-details/index.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+const NFTDetails = (props: Props) => {
+  const {
+    selectedAsset,
+    nftMetadata,
+    defaultCurrencies,
+    selectedNetwork
+  } = props
+
+  const onViewOnBlockExplorer = () => {
```
Comment (onyb) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-details/index.tsx:57]: Can we use `useExplorer` here? Maybe there's something to change in the hook itself.

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/accounts-and-transctions-list/index.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+  const getBalance = useBalance(selectedNetwork)
+
+  const findAccount = (address: string): WalletAccountType | undefined => {
+    return accounts.find((account) => address === account.address)
+  }
+
+  const accountsList = React.useMemo(() => {
+    if (selectedAsset?.isErc721) {
+      return accounts.filter((account) => Number(account.balance) === 1)
```
Comment (onyb) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/accounts-and-transctions-list/index.tsx:84]: Maybe we should check for non-zero balance here. WDYT?

**Review by onyb — APPROVED** [review]

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-details/index.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-details/index.tsx]: Updated and is now using the `useExplorer` hook!

**Pull request merged**